playwright/_impl/_frame.py

```diff
@@ -805,3 +847,20 @@ async def set_checked(
 
     async def _highlight(self, selector: str) -> None:
         await self._channel.send("highlight", {"selector": selector})
+
+    def _locals_to_params_with_timeout(self, args: Dict) -> Dict:
+        params = locals_to_params(args)
+        params["timeout"] = self._timeout(params.get("timeout"))
+        return params
+
+    def _locals_to_params_with_navigation_timeout(self, args: Dict) -> Dict:
+        params = locals_to_params(args)
+        params["timeout"] = self._navigation_timeout(params.get("timeout"))
+        return params
+
+    def _locals_to_params_without_timeout(self, args: Dict) -> Dict:
+        params = locals_to_params(locals())
+        # Timeout is deprecated and does nothing
+        if "timeout" in params:
```

Comment:
When do we end up in that if? I'd like to either not have this if or add a more detailed comment.